apps/server/src/database/prisma/repositories/prisma-locations-repository.ts

```diff
@@ -30,15 +30,41 @@ export class PrismaLocationsRepository implements ILocationsRepository {
 
     async deleteMany(locationsIds: string[]): Promise<void> {
         try {
-            await prisma.location.deleteMany({
+            for (const locationId of locationsIds) {
+            const location = await prisma.location.findUnique({
```

Comment:
Validating if the location exists in the repository exists is not necessary because you are already doing in the use case.